### PR TITLE
Synthetic Monitoring: Allow filtering out deprecated probes

### PR DESCRIPTION
**Pull request by julienduchesne**
I'd like to only use the current ones instead of the whole list

### CODE REVIEW

File: docs/data-sources/synthetic_monitoring_probes.md [hunk 1]
```diff
@@ -14,13 +14,18 @@ Data source for retrieving all probes.
...
+  filter_deprecated = true
+}
 ```
 
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
 ### Optional
 
+- **filter_deprecated** (Boolean) If true, only probes that are not deprecated will be returned. Defaults to `false`.
```
Comment (electron0zero) [inline docs/data-sources/synthetic_monitoring_probes.md]: I would lean to keeping it to `true` by default. why? ideally we should not default to scheduling checks to deprecated probes.
```suggestion
- **filter_deprecated** (Boolean) If true, only probes that are not deprecated will be returned. Defaults to `true`.
```

File: grafana/data_source_synthetic_monitoring_probes.go [hunk 2]
```diff
@@ -12,6 +12,12 @@ func dataSourceSyntheticMonitoringProbes() *schema.Resource {
 		Description: "Data source for retrieving all probes.",
 		ReadContext: dataSourceSyntheticMonitoringProbesRead,
 		Schema: map[string]*schema.Schema{
+			"filter_deprecated": {
+				Type:        schema.TypeBool,
+				Description: "If true, only probes that are not deprecated will be returned.",
+				Optional:    true,
+				Default:     false,
```
Comment (electron0zero) [inline grafana/data_source_synthetic_monitoring_probes.go]:
```suggestion
				Default:     true,
```

File: docs/data-sources/synthetic_monitoring_probes.md (hunk 1 above)
Comment (electron0zero) [reply docs/data-sources/synthetic_monitoring_probes.md]: deprecated probes will eventually be removed and turned off.

**Review by electron0zero — APPROVED** [review]
other then default comment, rest lgtm 👍🏼

File: docs/data-sources/synthetic_monitoring_probes.md (hunk 1 above)
Comment (julienduchesne) [reply docs/data-sources/synthetic_monitoring_probes.md]: Yeah, I was worried about unexpected changes to users of the provider but on the other hand, when they are removed, users will have changes in any case.

👍 will default to true

**Pull request merged**